### PR TITLE
Sprint 13

### PR DESCRIPTION
**Pull request by bmcorser**
Moving code from `leeloo` to `korben` where its responsibility is populating Django database or Elastic Search indexes with downloaded data from CH.
 - [x] Download CH ["free data"](http://download.companieshouse.gov.uk/en_output.html) to get CSVs locally
 - [x] Read CSVs, tranforming column names
 - [x] ~~In a transaction;~~ `TRUNCATE` the CH companies table and ~~use `COPY` to~~ dump CSVs there
 - [x] Do the equivalent in Elastic Search, with the additional logic of omitting those CH companies that are otherwise represented by CDMS records
 - [x] Finish writing ETL for CDMS → Django
 - [x] Initial population code for CDMS Django → ES
 - [x] Wait for models to get fixed in #5

### CODE REVIEW

File: korben/korben/sync/ch/download.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from lxml import etree
+import requests
+
+from . import constants
+from .. import utils as sync_utils
+
+
+def filenames():
+    'Get currently offered filenames from CH'
+    resp = requests.get(constants.DOWNLOAD_URL)
```
Comment (sdonk) [inline korben/korben/sync/ch/download.py]: What if this fails?

File: korben/korben/sync/ch/download.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    root = etree.parse(io.BytesIO(resp.content), parser).getroot()
+    retval = []
+    for anchor in root.cssselect('.omega a'):
+        retval.append(anchor.attrib['href'])
+    return retval
+
+
+def zips(names):
+    'Download zip files from CH with caching'
+    base_url = urllib.parse.urlparse(constants.DOWNLOAD_URL)
```
Comment (sdonk) [inline korben/korben/sync/ch/download.py]: Same as above

File: korben/korben/sync/ch/main.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    for key in CH_FIELDS:
+        ch_company[key] = row[key]
+    ch_company['incorporation_date'] = ch_date(row, 'incorporation_date')
+    return ch_company
+
+
+def main():
+    csv_paths = download.extract(download.zips(download.filenames()))
+    metadata = services.db.poll_for_metadata(config.database_url)
+    ch_company_table = metadata.tables['company_companieshousecompany']
```
Comment (sdonk) [inline korben/korben/sync/ch/main.py]: table names as constants?

File: korben/korben/sync/ch/download.py (hunk 2 above)
Comment (bmcorser) [inline korben/korben/sync/ch/download.py]: Just parsing the constant for a URL object here, but 👍 on the above.

Comment (sdonk) [inline korben/korben/sync/ch/download.py]: LOL, you're right. Sorry.

**Pull request merged**